File: etamen-backend/app/Modules/Health/Infrastructure/Models/HealthProfile.php
<?php

namespace App\Modules\Health\Infrastructure\Models;

use App\Models\User;
use App\Modules\Health\Domain\Enums\BloodType;
use App\Modules\Health\Domain\Enums\Gender;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class HealthProfile extends Model
{
    protected $fillable = [
        'patient_user_id',
        'date_of_birth',
        'gender',
        'height_cm',
        'weight_kg',
        'blood_type',
        'emergency_contact_name',
        'emergency_contact_phone',
        'notes',
    ];

    protected function casts(): array
    {
        return [
            'date_of_birth' => 'date',
            'gender' => Gender::class,
            'height_cm' => 'decimal:2',
            'weight_kg' => 'decimal:2',
            'blood_type' => BloodType::class,
        ];
    }

    public function patient(): BelongsTo
    {
        return $this->belongsTo(User::class, 'patient_user_id');
    }
}

File: etamen-flutter/lib/features/health/domain/entities/health_profile.dart
class HealthProfile {
  const HealthProfile({
    this.id,
    this.gender,
    this.birthDate,
    this.heightCm,
    this.weightKg,
    this.bloodType,
    this.chronicDiseases = const [],
    this.allergies = const [],
    this.currentMedications = const [],
    this.surgeries = const [],
    this.goals = const [],
    this.emergencyContactName,
    this.emergencyContactPhone,
    this.notes,
    this.createdAt,
    this.updatedAt,
  });

  final int? id;
  final String? gender;
  final String? birthDate;
  final String? heightCm;
  final String? weightKg;
  final String? bloodType;
  final List<String> chronicDiseases;
  final List<String> allergies;
  final List<String> currentMedications;
  final List<String> surgeries;
  final List<String> goals;
  final String? emergencyContactName;
  final String? emergencyContactPhone;
  final String? notes;
  final DateTime? createdAt;
  final DateTime? updatedAt;
}
